wrappers/react/src/hotTableContext.tsx

```diff
@@ -123,28 +123,31 @@ const HotTableContextProvider: React.FC<PropsWithChildren> = ({ children }) => {
   }, [])
 
   const componentRendererColumns = useRef<Map<number | 'global', boolean>>(new Map());
+
   const editorCache = useRef<HotEditorCache>(new Map());
   const renderedCellCache = useRef<Map<string, HTMLTableCellElement>>(new Map());
   const clearRenderedCellCache = useCallback(() => renderedCellCache.current.clear(), []);
   const portalCacheArray = useRef<React.ReactPortal[]>([]);
 
-  const getRendererWrapper = useCallback((rendererElement: React.ReactElement): typeof Handsontable.renderers.BaseRenderer => {
+  const getRendererWrapper = useCallback((Renderer: React.ComponentType<HotRendererProps>): typeof Handsontable.renderers.BaseRenderer => {
     return function (instance, TD, row, col, prop, value, cellProperties) {
-      if (renderedCellCache.current.has(`${row}-${col}`)) {
-        TD.innerHTML = renderedCellCache.current.get(`${row}-${col}`)!.innerHTML;
+      const rowColKey = `${row}-${col}`
+      if (renderedCellCache.current.has(rowColKey)) {
+        TD.innerHTML = renderedCellCache.current.get(rowColKey)!.innerHTML;
       }
 
       if (TD && !TD.getAttribute('ghost-table')) {
-
-        const {portal, portalContainer} = createPortal(rendererElement, {
-          TD,
-          row,
-          col,
-          prop,
-          value,
-          cellProperties,
-          isRenderer: true
-        }, TD.ownerDocument);
+        const rendererElement = (
+            <Renderer instance={instance}
+                      TD={TD}
+                      row={row}
+                      col={col}
+                      prop={prop}
+                      value={value}
+                      cellProperties={cellProperties}/>
+        );
+
+        const {portal, portalContainer} = createPortal(rendererElement, `${rowColKey}-${Math.random()}`, TD.ownerDocument);
```

Comment:
We should be able to remove the Math.random() such as `${rowColKey}-${instanceGuid}`